app/controllers/finders_controller.rb

```diff
@@ -20,9 +20,9 @@ def show
         @search_query = initialize_search_query
         if content_item.is_search? || content_item.is_finder?
           @spelling_suggestion_presenter = spelling_suggestion_presenter
-          render json: json_response
+          render json: json_response, status: response_status
```

Comment:
Hey @bilbof If we raise a 400 the JSON fetch fails :-( Is there another way I can log that the parameters are bad other than by using a status code?

Reply:
Would statsd be a viable alternative?

Reply:
I think that's probably the result of the current Ajax error handling:

```
      return $.ajax({
        url: this.action,
        data: this.state,
        searchState: searchState
      }).done(function (response) {
        liveSearch.cache($.param(liveSearch.state), response)
        liveSearch.displayResults(response, this.searchState)
      }).error(function () {
        liveSearch.showErrorIndicator()
      })
```

In the `.error` callback you could handle the 400 status code, and check whether there's an error message to be rendered for an input field.

Reply:
Thanks @bilbof 

I discussed that option with @kr8n3r earlier and the concern was that there might be legitimate reasons for a 400 meaning we shouldn't do the fetch.

Maybe I could raise a more specific 4xx and handle just that one type?

Reply:
Oh sorry @bilbof I misread your comment. Yep good idea thank you.

Reply:
In the error callback, it's mostly been handling 500 errors from search-api timing out. So you'd need to change this to handle the 400 error and keep the current behaviour for other status codes.

That might look like this:

```
      }).error(function (response) {
        if (response.status == 400) { return handleInputError() }
        liveSearch.showErrorIndicator()
      })
```

Is there a more specific error code you had in mind?

Reply:
at the moment we provide a json fail regardless of the 400 error, as the results still come back from the api
if we were to read the error message for the field in the JSON, we'd still need the status to be 200.

unless i'm understanding the requirement incorrectly

Reply:
The response body can still be JSON even if the status code is not 200. If the status code is 400 jQuery will call the `error` callback, so `error` handles the response. Within the `error` callback you can treat the response in the same way as you treat it in the `success` handler.

Reply:
sure, we can do that. let me think how to handle that and other 400 cases

Reply:
i'm now battling with the state not being updated

Reply:
Does that matter? It's updating if it's valid. Maybe that's fine??

Reply:
when firing 400 status, ajax does not return a new state back so the function that updates the url does not fire for this or subsequent (valid) ajax requests until the fields with the error is fixed, leaving a stale url

Reply:
@kr8n3r 

Following pairing effort with @bilbof it looks like error handling is happening [upstream in Slimmer](https://github.com/alphagov/slimmer/blob/b56bd87a09824262aac298490e0d14122c86df5c/lib/slimmer/app.rb#L85) 
ie when we return a 400 if the search query is invalid, slimmer decides that we show the default gov.uk 400 error page. We can skip slimmer, but then we also lose the headers and formatting of the page that slimmer brings over from static. 

Decision: Return 200's for both the application and JSON response.